graylog2-server/src/main/java/org/graylog2/inputs/codecs/JsonPathCodec.java

```diff
@@ -105,6 +141,53 @@ protected String buildShortMessage(Map<String, Object> fields) {
         return shortMessage.toString();
     }
 
+    public Map<String, Object> flatten(String json) throws JsonTypeException {
+        try {
+            return flatten("", objectMapper.readTree(json));
+        } catch (IOException e) {
+            LOG.warn("IOException occurred", e);
+        }
+        return Map.of();
+    }
+
+    private Map<String, Object> flatten(String currentPath, JsonNode jsonNode) throws JsonTypeException {
+        if (jsonNode.isObject()) {
+            ObjectNode objectNode = (ObjectNode) jsonNode;
+            Iterator<Map.Entry<String, JsonNode>> iter = objectNode.fields();
+            String pathPrefix = currentPath.isEmpty() ? "" : currentPath + ".";
+            ImmutableMap.Builder<String, Object> builder = ImmutableMap.builder();
+
+            while (iter.hasNext()) {
+                Map.Entry<String, JsonNode> entry = iter.next();
+                builder.putAll(flatten(pathPrefix + entry.getKey(), entry.getValue()));
+            }
+            return builder.build();
+        } else if (jsonNode.isArray()) {
+            ArrayNode arrayNode = (ArrayNode) jsonNode;
+            ImmutableMap.Builder<String, Object> builder = ImmutableMap.builder();
+            for (int i = 0; i < arrayNode.size(); i++) {
+                builder.putAll(flatten(currentPath + i, arrayNode.get(i)));
+            }
+            return builder.build();
+        } else if (jsonNode.isTextual()) {
+            TextNode textNode = (TextNode) jsonNode;
+            return Map.of(currentPath, textNode.toString());
+        } else if (jsonNode.isNumber()) {
+            NumericNode numericNode = (NumericNode) jsonNode;
+            return Map.of(currentPath, numericNode.numberValue());
+        } else if (jsonNode.isBoolean()) {
+            BooleanNode booleanNode = (BooleanNode) jsonNode;
+            return Map.of(currentPath, booleanNode.asBoolean());
+        } else {
+            throw new JsonTypeException("Warning: JSON contains type not supported by the flatten method.");
```

Comment:
Could we add some information about the node type, its name or something helpful for debugging, if this situation occurs?